src/pip/_internal/configuration.py

```diff
@@ -292,6 +292,20 @@ def _construct_parser(self, fname):
                     "Configuration file contains invalid %s characters.\n"
                     "Please fix your configuration, located at %s\n"
                 ) % (locale.getpreferredencoding(False), fname))
+            except configparser.MissingSectionHeaderError as e:
+                raise ConfigurationError(
+                    "ERROR: "
+                    "Configuration file headers cannot be parsed. \n"
+                    "Please fix the section formatting of the "
+                    "configuration located at line %s of file %s\n" % (e.lineno, fname)
+                )
+            except configparser.ParsingError as e:
+                raise ConfigurationError(
+                    "ERROR: "
+                    "Configuration file cannot be parsed. \n"
+                    "%s\n" % e.message
```

Comment:
This line is killing the CI build (no "message" attribute), but I can't seem to reproduce it. I suspect something about what I'm doing is bad practice...?